mzLib/Omics/BioPolymer/VariantApplication.cs

```diff
@@ -121,188 +212,388 @@ public static List<TBioPolymerType> ApplyVariants<TBioPolymerType>(TBioPolymerTy
+                variantSeq,
+                variantGettingApplied.Description,
+                vcfDescription,
+                variantModDict.Count == 0 ? null : variantModDict);
+
+            bool intersectsAppliedRegionIncompletely = protein.AppliedSequenceVariations
+                .Any(x => variantGettingApplied.Intersects(x) && !variantGettingApplied.Includes(x));
+
             IEnumerable<SequenceVariation> appliedVariations = new[] { variantAfterApplication };
-            string seqAfter = null;
-            if (intersectsAppliedRegionIncompletely)
-            {
-                // use original protein sequence for the remaining sequence
-                seqAfter = protein.BaseSequence.Length - afterIdx <= 0 ? "" : protein.ConsensusVariant.BaseSequence.Substring(afterIdx);
-            }
-            else
+            if (!intersectsAppliedRegionIncompletely)
             {
-                // use this variant protein sequence for the remaining sequence
-                seqAfter = protein.BaseSequence.Length - afterIdx <= 0 ? "" : protein.BaseSequence.Substring(afterIdx);
                 appliedVariations = appliedVariations
                     .Concat(protein.AppliedSequenceVariations.Where(x => !variantGettingApplied.Includes(x)))
                     .ToList();
             }
-            string variantSequence = (seqBefore + seqVariant + seqAfter).Split('*')[0]; // there may be a stop gained
+            else
+            {
+                seqAfter = afterIdx >= protein.ConsensusVariant.BaseSequence.Length
+                    ? string.Empty
+                    : protein.ConsensusVariant.BaseSequence.Substring(afterIdx);
+            }
+
+            string newBaseSequence = (seqBefore + variantSeq + seqAfter).Split('*')[0];
+
+            var adjustedProteolysisProducts =
+                AdjustTruncationProductIndices(variantAfterApplication, newBaseSequence, protein, protein.TruncationProducts);
+
+            // AdjustModificationIndices merges existing protein-level mods and variant-specific mods (promotion to applied isoform)
+            var adjustedModifications =
+                AdjustModificationIndices(variantAfterApplication, newBaseSequence, protein);
+
+            var adjustedAppliedVariations =
+                AdjustSequenceVariationIndices(variantAfterApplication, newBaseSequence, appliedVariations);
+
+            // Centralized creation to ensure AppliedSequenceVariations are wired on the new variant
+            var created = BuildVariant(
+                protein,
+                newBaseSequence,
+                adjustedAppliedVariations,
+                adjustedProteolysisProducts,
+                adjustedModifications,
+                individual);
+
+            // Normalize UniProt sequence attributes (length + mass)
+            try
+            {
+                var seq = created?.BaseSequence;
+                if (!string.IsNullOrEmpty(seq))
+                {
```

Comment:
Inside this try statement, you're using reflection to access properties of your variant. This is concerning, as reflection should rarely be used in production code. It's slow and finicky.

Can you refactor this to avoid using reflection?